Nef_2/include/CGAL/Nef_2/PM_io_parser.h

```diff
@@ -267,14 +276,32 @@ bool PM_io_parser<PMDEC>::read_face(Face_handle f)
   int n, ei, vi; Mark m;
   if ( !(in >> n) || !check_sep("{") ) return false;
   if ( !(in >> ei) || !check_sep(",") ) return false;
-  if (ei >= 0) f->set_halfedge(Halfedge_of[ei]);
+  if (ei >= 0 && ei < en)
+  {
+    f->set_halfedge(Halfedge_of[ei]);
+  }
+  else
+  {
+    in.clear(std::ios_base::badbit);
```

Comment:
I would prefer:
```suggestion
    in.setstate(std::ios_base::badbit);
```
I find strange to call `clear` to set a bit, and `clear` also has the defect that it clears all the other bits.